### PR TITLE
Write documentation

### PR DESCRIPTION
**Pull request by 0x26rew**
Please read the documentation and give some advice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,120 @@
-# FPGA-TrafficLight
\ No newline at end of file
+# FPGA-TrafficLight
```
Comment (choucl) [inline README.md:1]: Duplicate of the h1 header

File: README.md [hunk 2]
```diff
@@ -1 +1,120 @@
...
+    <tr>
+        <td>11</td>
+        <td>White</td>
+        <td>Display the duration of <b>red</b> lights</td>
+        <td>Adjust the duration of <b>red</b> lights</td>
+    </tr>
+</table>
+
+
+A debounce circuit is used here for `BTN` .
```
Comment (choucl) [inline README.md]: Maybe we can detail how we implement the denounce circuit here

File: README.md [hunk 3]
```diff
@@ -1 +1,120 @@
...
+
+### Modules
+
+1. **Controller** module
+
+   Major part in this project, controlling the state transition and handling user input(switch and button).
```
Comment (choucl) [inline README.md]: Maybe we should add more details here, like which parts are sequential logics and which parts are combinational circuits, also, how they collaborate with each other’s?

File: README.md [hunk 4]
```diff
@@ -1 +1,120 @@
...
+| ------ | ------ |
+| 00     | White  |
+| 01     | Red    |
+| 10     | Green  |
+| 11     | Yellow |
+
+### Modules
+
+`Decoder.v` is the only module used in this project which maps the input switch to output RGB color. 
+
```
Comment (choucl) [inline README.md:27]: We can simply add how we decode the signal from the input. Like: “We decoded the input signal using a `case` statement. According to the input we directly output the corresponding to the RGB LEDs”

File: README.md [hunk 5]
```diff
@@ -1 +1,120 @@
...
+<div align="left">
+    <img src="images/traffic_light.png" width=75% height=75%>
+</div>
+
+
+## Discussions
+
+1. Why add `blinky.xdc`?
+   The `create_clock` command is used to define the primary clock of our design, and the `create_generated_clock` command defines the generated clock that follows the master clock, which is the primary clock in our design.
```
Comment (choucl) [inline README.md:112]: The first problem should have one more extra line between the problem and answer, or the answer would be directly concatenated after the the problem when rendering

**Review by choucl — APPROVED** [review]

**Review by jumha — APPROVED** [review]

**Pull request merged**